dashboard/src/actions/auth.ts

```diff
@@ -39,3 +51,20 @@ export function logout(): ThunkAction<Promise<void>, IStoreState, null, AuthActi
     dispatch(setAuthenticated(false));
   };
 }
+
+export function tryToAutoAuthenticate(): ThunkAction<Promise<void>, IStoreState, null, AuthAction> {
+  return async dispatch => {
+    dispatch(autoAuthenticating());
+    try {
+      const token = await Auth.fetchToken();
+      if (token) {
+        Auth.setAuthToken(token);
```

Comment:
Are we validating this token against the API server? See `Auth.validateToken(token);`

Note that a lot of the logic is shared with `authenticate`